ledger/internal/eval.go

```diff
@@ -157,12 +157,14 @@ func (x *roundCowBase) getCreator(cidx basics.CreatableIndex, ctype basics.Creat
 		return foundAddress.address, foundAddress.exists, nil
 	}
 
-	address, exists, err := x.l.GetCreatorForRound(x.rnd, cidx, ctype)
+	address, exists, rnd, err := x.l.GetCreator(cidx, ctype)
 	if err != nil {
 		return basics.Address{}, false, fmt.Errorf(
 			"roundCowBase.getCreator() cidx: %d ctype: %v err: %w", cidx, ctype, err)
 	}
-
+	if exists && rnd != x.rnd {
```

Comment:
This condition here is true for some cases, but false when roundCowBase has the round `round-1` in `MakeDebugBalances`.